sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -148,6 +148,15 @@ case class InsertIntoHiveTable(
            |Table partitions: ${table.partitionColumnNames.mkString(",")}""".stripMargin)
     }
 
+    lazy val oldPart = if ( numDynamicPartitions == 0) {
```

Comment:
if it's a lazy val, do we still need the `if`?

Reply:
+1